### PR TITLE
fix(types, helpers)!: EditMessage helper transformer & change `Embed` to `DiscordenoEmbed`

### PR DESCRIPTION
**Pull request by lts20050703**
Note: I changed `transformer/embed.ts` because I looked at `helpers/messages/editMessage.ts` and see `provider: embed.provider` `fields: embed.fields` because both of them's properties are only one word therefore no camelCase to snake_case manual conversion needed

### CODE REVIEW

File: transformers/embed.ts [hunk 1]
```diff
@@ -42,12 +42,7 @@ export function transformEmbed(bot: Bot, payload: SnakeCasedPropertiesDeep<Embed
         width: payload.video.width,
       }
       : undefined,
-    provider: payload.provider
```
Comment (itohatweb) [inline transformers/embed.ts:45]: Unrelated change please do this in a different PR.
I want to stick to the 1 PR 1 change schema.

**Review by itohatweb — CHANGES_REQUESTED** [review]

**Review by Skillz4Killz — APPROVED** [review]
